ClosedXML/Excel/Ranges/XLRanges.cs

```diff
@@ -256,5 +261,15 @@ public IXLRanges Consolidate()
             var engine = new XLRangeConsolidationEngine(this);
             return engine.Consolidate();
         }
+
+        public void CalculateMergedCells()
+        {
+            _mergedCellAddresses = new HashSet<IXLAddress>(_ranges.Where(r => r.RangeAddress.IsValid).SelectMany(r => r.Cells().Select(c => c.Address)));
```

Comment:
This will cause a huge memory allocation for large merged ranges. Consider this:
```
            using (var wb = new XLWorkbook())
            {
                var ws = wb.Worksheets.Add("Sheet1");
                ws.Range("A:C").Merge();
                ws.Range("A1:C1").Value = "ASDFGHJ";
                ws.Columns("A:C").AdjustToContents();
            }
```
With this PR it takes more than 7 seconds to execute, and cost 1 GB of RAM while without the PR it takes only about 0.7 sec.

On the other hand, having many small merged ranges is much more common than having a few huge ones. So maybe it's fine.

Reply:
Good point(s). I'll give it a thought.